### DISCUSSION TITLE
Should we support languages no one uses?

### DISCUSSION THREAD

**Discussion by Felienne** [hedyorg/hedy#5604] [Language Ideas]
We have a few language translations that are not people's native language or culture, such as Interlingua, Esperanto, and recently someone added Middle English. I don't know how I feel about this, any language we add adds maintenance effort, and these languages do not really seem to add to our vision?

I am open to hear about what other people think!

**Comment by Felienne** [comment]
RelatedL in https://github.com/hedyorg/hedy/pull/5607 Old English was added but that leads to issues so I removed it in Weblate.

**Comment by jschwartzenberg** [comment]
Are some languages handled more or less automated? I do see risks if no speakers of the language are involved.

I was just looking at Ukrainian for example and the translation of the name as it is currently done is something that would likely come out of an automated translator influenced by Russian.

Note that the current translated name there is "Хеді". Which in Dutch would sounds like "Chedi" (soft g).
A translation that would match in pronunciation would be "Геді".

**Reply by Felienne** [reply]
Hi @jschwartzenberg!

It is very hard for us to keep track of the quality of translations, so I would say your best angle here, if you want to help, is to improve the translations. If you need help doing that, let us know!

**Reply by jschwartzenberg** [reply]
@Felienne, thank you for the quick response!!
Yeah it would be interesting to look into helping out. I had been looking at the commits and I couldn't really figure out whether translations where automated in some way or whether they were done by hand.

I should note that my native language is also Dutch, not Ukrainian, but I do speak Ukrainian daily, so I could possibly contribute. Or maybe it would make sense to first get Dutch up to 100%?

Is there a guide I can follow with regards to translating? Does it make sense to find reference material? I know a few other programming languages also exist in non-English languages and I do have a book in Ukrainian on learning Python.

**Reply by Felienne** [reply]
We don't have a system for automatical translation, not do we encourage it, but we can't see what people use in doing theirs.

Info on how to translate is here: https://github.com/hedyorg/hedy/wiki/Hedy-Translation-Tutorial

You can also join the Discord where you can ask other translators how they made choices, and yes looking at what other books or systems use is a common source of information. 

Happy to help if you have more questions!

**Reply by jschwartzenberg** [reply]
I just signed up to Weblate.

Recently I tried to join Discord, but I was unable to get through all their validation. Is there another mechanism, possibly some kind of bridge, that could enable me to reach people there?

**Reply by AnneliesVlaar** [reply]
Hi @jschwartzenberg! Sad that you couldn't join our Discord server. As far as I am aware there is not another easy way to join other than the invite link. I just checked and in Discord there are at the moment no other people that set Ukrainian as their language.

**Reply by jschwartzenberg** [reply]
@AnneliesVlaar thank you for checking and responding!! Sadly many services nowadays have strong filtering in place from large parts of the world compared to countries such as NL. I could possibly try again to do it from a Dutch IP address, but I guess many people won't have this option. Has it been considered to look at something like Matrix? The University of Twente has an instance that anybody can use (https://chat.utwente.io/), maybe the VU also has something like this?

I hope to give Hedy a try soon in a few difference languages :)
